### PR TITLE
[web] Long press fix on Safari on IOS

### PR DESCRIPTION
**Pull request by nturgut**
The long press in IOS does not bring the virtual keyboard but still focus on the dom element. After this if a user taps on the input element the virtual keyboard will shift the page up. (the video I recorded turned out to be to big for github so I am not adding it here.)

in order to fix this, i added one more listener. this time for the click. this listener carries the input field temporarily out of the visible page to prevent shifting.

fixing eap issue: https://github.com/flutter/flutter/issues/41638

### CODE REVIEW

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart [hunk 1]
```diff
@@ -580,6 +581,43 @@ class IOSTextEditingStrategy extends DefaultTextEditingStrategy {
...
+  /// placed to its correct position after [_delayBeforePositioning].
+  void _addTapListener() {
+    _subscriptions.add(domElement.onClick.listen((_) {
+      // Check if the element is already positioned. If not this does not fall
+      // under `The user was using the long press, now they want to enter text
+      // via keyboard` journey.
+      if (_canPosition) {
+        // Re-place the element somewhere outside of the screen.
+        domElement.style.transform = 'translate(-9999px, -9999px)';
+        _canPosition = false;
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/text_editing/text_editing.dart]: Suggestion: we could call `initializeElementPosition()` here.

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart [hunk 2]
```diff
@@ -580,6 +581,43 @@ class IOSTextEditingStrategy extends DefaultTextEditingStrategy {
...
+
+        // Re-configure the timer to place the element.
+        _positionInputElementTimer?.cancel();
+        _positionInputElementTimer = Timer(_delayBeforePositioning, () {
+          _canPosition = true;
+          positionElement();
+        });
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/text_editing/text_editing.dart]: The exact same block of code is repeated in the onFocus listener and I'm concerned about the two getting out of sync eventually. Could we maybe move this code into a method (e.g. `schedulePositioning()`)?

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/lib/src/engine/text_editing/text_editing.dart]: done. thanks!

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart (hunk 2 above)
Comment (nturgut) [reply lib/web_ui/lib/src/engine/text_editing/text_editing.dart]: done.

**Comment by nturgut** [comment]
I am merging since the breakage is in Fushia (unrelated to the web engine)

**Pull request merged**